lib/logstash/inputs/example.rb

```diff
@@ -28,11 +28,17 @@ def register
   end # def register
 
   def run(queue)
+    @thread = Thread.current
+
     Stud.interval(@interval) do
       event = LogStash::Event.new("message" => @message, "host" => @host)
       decorate(event)
       queue << event
+      break if stop?
```

Comment:
i wonder if break is necessary here? what are the benefits of having it (I mean educationally for people seeing this examples?)